src/app/accounts/change-password.component.ts

```diff
@@ -76,16 +79,29 @@ export class ChangePasswordComponent extends BaseChangePasswordComponent {
         }
     }
 
-    async performSubmitActions(newMasterPasswordHash: string, newKey: SymmetricCryptoKey,
-        newEncKey: [SymmetricCryptoKey, CipherString]) {
-        const setRequest = new SetPasswordRequest();
-        setRequest.newMasterPasswordHash = newMasterPasswordHash;
-        setRequest.key = newEncKey[1].encryptedString;
+    async performSubmitActions(masterPasswordHash: string, key: SymmetricCryptoKey,
+        encKey: [SymmetricCryptoKey, CipherString]) {
+        const request = new SetPasswordRequest();
+        request.masterPasswordHash = masterPasswordHash;
+        request.key = encKey[1].encryptedString;
+        request.masterPasswordHint = this.hint;
+        request.kdf = this.kdf;
+        request.kdfIterations = this.kdfIterations;
+
+        const keys = await this.cryptoService.makeKeyPair(encKey[0]);
+        request.keys = new KeysRequest(keys[0], keys[1].encryptedString);
```

Comment:
I can probably extract our set password componeent out to jslib to use as a base

Reply:
Updated with latest changes.